v2/pkg/genruntime/enum.go

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package genruntime
+
+import "strings"
+
+// ToEnum does a case-insensitive conversion of a string to an enum using a provided conversion map.
+// If the required value is not found, a literal cast will be used to return the enum.
+func ToEnum[T ~string](str string, enumMap map[string]T) T {
```

Comment:
Very cool

Reply:
Yeah. Full credit though - you gave me the method signature. :grin: